### ISSUE TITLE
Represent enzymatic complex enablers according to GO-CAM spec

### ISSUE THREAD

**Issue by thomaspd** [geneontology/pathways2GO#302]
For Reactome complexes that are controllers of enzymatic reactions, these should be aligned with GO-CAM to specify gene product IDs and not PRO IDs. To do this, we will need to handle two different cases:
1) If Reactome specifies the catalytic subunit, the enabler of the reaction should be the catalytic subunit and not the complex. The rest of the complex should be ignored for now.
2) If Reactome does not specify the catalytic subunit, the enabler of the reaction should be a GO protein-containing complex instance (GO:0032991) with has_part links to each protein subunit. Small molecule components of complexes should be ignored for now. So if a Reactome complex is composed of just one gene product and one or more small molecules, then it should be treated the same as case #1 above and connect the activity directly to the gene product without a protein-containing complex individual.

**Comment by deustp01** [comment]
Yes! We have treated case 2 as obviously true for a long time, but I'm not sure the code to enable it has ever been implemented in the GO-CAM conversion process. And when it is implemented, it should also generate a report of all the catalystActivity instances that it fixed in this way, to be fed back to Reactome to patch the Reactome annotations that are the single source of truth here. @dustine32 ?

**Comment by dustine32** [comment]
@deustp01 Yes, we can log out the number 2 cases where the resulting "complex" only has a single `has_part` GP.

@thomaspd If you can find an example complex with the active unit specified that would help. I'll keep looking too.

**Comment by dustine32** [comment]
Also, asking @huaiyumi for any examples of active subunit annotation in Reactome that I can look for in the BioPAX.

**Comment by deustp01** [comment]
But meanwhile, here is a short arbitrary list of catalystActivity instances whose physicalEntity is a heteromeric complex and whose activeUnit is a protein monomer
[](https://reactome.org/content/schema/instance/browser/1806156) 
[](https://reactome.org/content/schema/instance/browser/5676051 )
https://reactome.org/content/schema/instance/browser/6798176 
https://reactome.org/content/schema/instance/browser/1806283
https://reactome.org/content/schema/instance/browser/8868073 
https://reactome.org/content/schema/instance/browser/5358378 
https://reactome.org/content/schema/instance/browser/109879 
https://reactome.org/content/schema/instance/browser/9836928 
Each URL points to a page that lists the names and dbIDs of the heteromeric complex, the protein monomer activeUnit, and the reaction that the caqtalystActivity mediates.
I can also make a list of samples of catalystActivity instances where the physicalEntity is a set of heteromeric complexes and the activeUnit is a set of monomers or a set of subcomplexes, also of cases where the heteromeric complex involves both protein and non-protein (RNA or DNA or small-molecule) subunits, if any of those are of interest.

**Comment by ukemi** [comment]
@dustine32 Does the catalyst activity here help? R-HSA-21271

**Comment by deustp01** [comment]
> is embedded in a comment field

If I understand what you're saying correctly, yes, if you look at the instancebrowser view for the EWAS [PI4KB [Golgi membrane] (Homo sapiens)](https://reactome.org/content/detail/R-HSA-1806202) its role as the activeUnit of a complex involved in catalysis is shown as a comment. But if you come at the annotation from the other direction - start with the catalystActivity instance [1-phosphatidylinositol 4-kinase activity of ARF1/3:GTP:PI4KB [Golgi membrane]](https://reactome.org/content/schema/instance/browser/1806156) then its role is shown as an attribute. Or am I misunderstanding the problem?

Also, it makes sense to me to work starting from reactions that have associated catalystActivities, systematically looking at what the physicalEntity of each catalystActivity is, and if that physicalEntity is not at EWAS or set of EWASs, then proceed further to see if it fits case 2 above.

Also also, if a by-product of this survey were a list of catalystActivites where the physicalEntity is a complex or set of complexes but the activeUnit slot is null, that list would be the starting point for re-curation to fill the empty slots. And if in each case the components of the complex could be checked in the central GO annotation file to see if any have been assigned the same GO molecular function as Reactome has assigned to the whole complex, that would make the re-curation process at Reactome much faster and more reliable. @ukemi I know we talked about something like this with Ben Good; I don;t know how close he got to implementing it.

**Comment by deustp01** [comment]
> And if in each case the components of the complex could be checked in the central GO annotation file to see if any have been assigned the same GO molecular function as Reactome has assigned to the whole complex, that would make the re-curation process at Reactome much faster and more reliable. @ukemi I know we talked about something like this with Ben Good; I don;t know how close he got to implementing it.

That failed - in many cases the catalystActivity of the whole complex has been assigned to all of its protein components, but perhaps re-doing it with the cleaned-up fly set of complex component functions would yield good results.

**Comment by deustp01** [comment]
Summarizing the discussion so far as a to-do list.

- Identify cases where Reactome has assigned a catalystActivity to an entire heteromeric complex but experimental data identify one of the gene product components of the complex as capable of the catalystActivity. Annotate that component of the complex as its activeUnit.
- In the case of a homomeric complex, creation of activeUnit annotations is formally redundant, but is it useful to support accurate parsing of Reactome BioPAX entries into GO-CAM models?
- In cases where the catalystActivity is an emergent property of two or more gene product components of the complex, [current GO documentation](https://wiki.geneontology.org/Contributes_to) suggests that correct GO annotation is to assert that all of these components "contribute to" the catalystActivity. In Reactome, the activeUnit attribute of a catalystActivity instance can be multivalued. Could we enforce a new curation standard in Reactome, connected to a new rule for parsing BioPAX into GO-CAM, that if the activeUnit slot has a single gene product entry, that gene product "enables" the catalystActivity while if the slot has more than one value, each of those gene products "contributes to" the molecular activity.
- To annotate regulatory subunits of multimeric complexes, Reactome allows creation of regulation instances whose physicalEntity is a complex and whose [regulatory] activeUnit is a gene product subunit of the complex, so regulatory abilities of individual subunits of the catalytic complex could be annotated just as catalytic abilities are, and emergent regulatory abilities of two or more subunits could be annotated just as emergent catalytic activities are.
- In all of this, can the cleaned-up of annotations of specific functions to specific components of Drosophila complexes (previous comment) be useful to predict specific functions of specific components of the homologous human complexes?

**Comment by deustp01** [comment]
> @deustp01 Yes, we can log out the number 2 cases where the resulting "complex" only has a single `has_part` GP.

@dustine32 @ukemi just to document the current state / need, here is an active unit in the "carnitine biosynthesis" pathway http://noctua.geneontology.org/editor/graph/gomodel:R-HSA-71262 where the physical entity is a complex involving one copy of one gene product and one copy each of a couple of chemical entities. Can the GO-CAM generation script be re-done to identify the gene product and make it the enabler?

<img width="189" alt="Screenshot 2023-11-14 at 5 47 50 PM" src="https://github.com/geneontology/pathways2GO/assets/12737623/c5f83e85-ef55-4e0e-98ea-77d36a4e5c31">

Or if that is hard or dangerous, can we plan to generate a list (partial is OK to start) of the number 2 cases, that we can use to figure out how to bulk-edit the Reactome annotations to add the missing activeUnit annotations, so that the existing GO-CAM genertion script can use them? (A practical issue here is whether David and I, as we (re)curate pathways should add this information manually as part of our work, or leave it out because a script will soon be available to do it automatically?

**Comment by ukemi** [comment]
@dustine32 I think this proposal is directly in line with what we had talked about on the call today. I think you have already done it for when there is a single GP as the enabler, but we should do it with the complexes too. For these cases, would it be possible to use the UniProt GCRP identifiers instead of a REACTO id? We need to start weeding away the Reacto identifiers.

**Comment by ukemi** [comment]
Hi @dustine32. Note that part of the requirement in the second point of this post is to ignore small molecules. 

> If Reactome does not specify the catalytic subunit, the enabler of the reaction should be a GO protein-containing complex instance (GO:0032991) with has_part links to each protein subunit. Small molecule components of complexes should be ignored for now. So if a Reactome complex is composed of just one gene product and one or more small molecules, then it should be treated the same as case 1 above and connect the activity directly to the gene product without a protein-containing complex individual.

This is consistent with the discussion in #327.
